ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,251 @@
+    ->fields('d', array('id', 'time_start', 'status', 'text'))
+    ->orderBy('id', 'DESC')
+    ->execute()
+    ->fetchAll();
+  $max_count = count($report);
+  if (!$max_count) {
+    return drupal_set_message(t('No cron report.'), 'warning');
+  }
+  drupal_add_css(drupal_get_path('module', 'ding_dibs') . '/css/ding_dibs.css');
+  $header = array(array('data' => t('Message'), 'class' => 'payment'));
+  $rows = array();
+  foreach ($report as $record) {
+    if (empty($record->text)) {
+      $day = format_date($record->time_start, 'custom', 'd', date_default_timezone_get());
+      $hour = format_date($record->time_start, 'custom', 'H:i', date_default_timezone_get());
+      $month = format_date($record->time_start, 'custom', 'F', date_default_timezone_get());
+      $record->text = $day . 'th ' . $month . ' ' . $hour . ': Job failed.';
+    }
+    $row = array(
+      array(
+        'data' => t($record->text),
+        'class' => 'payment',
+      ),
+    );
+    $rows[] = $row;
+  }
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+  ));
+  return $output;
+}
+
+/**
+ * The function adds to the dataset for the numbers zero to ten.
+ *
+ * @param $item
+ *   An array of input data for the time period.
+ *
+ * @return
+ *   Returns an array with the format of the time 01, 02.
+ */
+function _ding_dibs_generate_zero($item) {
+  return str_pad($item, 2, "0", STR_PAD_LEFT);
+}
+
+/**
+ * The function generates a table with data reporting database table
+ * so dibs transactions.
+ *
+ * @param $dibs_transactions_value
+ *   Object database table data dibs transactions
+ *
+ * @param $order_info_long
+ *   Long description of table warrant dibs_transactions.
+ *
+ * @param $status
+ *   Status display last column of the table report
+ *
+ * @param $dibs_transactions_value->payment_order_id
+ *   ID orders payment.
+ *
+ * @param $dibs_transactions_value->payment_transaction_id
+ *   ID orders transaction.
+ *
+ * @param $dibs_transactions_value->payment_time
+ *   Time payment.
+ *
+ * @param $dibs_transactions_value->customer_uid
+ *   User ID who has charge.
+ *
+ * @param $dibs_transactions_value->payment_price
+ *   Product price the order in payment.
+ *
+ * @param $dibs_transactions_value->payment_status
+ *   Status payment.
+ *
+ * @param $dibs_transactions_value->provider_payment_status
+ *   Provider payment status.
+ *
+ * @param $dibs_transactions_value->dibs_capture_status
+ *   Captured payment status.
+ *
+ * @return
+ *   Return value table dibs_transactions for mailing reports.
+ */
+function _ding_dibs_report_generation_values($dibs_transactions_value) {
+  $status = $dibs_transactions_value->payment_status . $dibs_transactions_value->provider_payment_status . $dibs_transactions_value->dibs_capture_status;
+  if ($status == 100) {
+    $status = 'warning';
+  }
+  else {
+    $status = 'error';
+  }
+  $order_info_long = unserialize($dibs_transactions_value->order_info_long);
```

Comment:
Please add a code comment explaining what is going on here and why. The extraction and manipulation of $order_info_long is not selfdocumenting.